components/builder-depot/src/backend/s3.rs

```diff
@@ -0,0 +1,347 @@
+                ident.release().unwrap(),
+                target.architecture,
+                target.platform
+            ))
+    }
+
+    // Helper function to validate whether or not a package
+    // exists in the configured bucket.
+    // This magical incantation was difficult to find. We've
+    // removed the use of this function for now, I've left it
+    // here for any achaeologist that might need it should we
+    // decide this validation is useful in the future.
+    //pub fn exists<T>(&self, ident: &T, target: &PackageTarget) -> Result<bool>
+    //where
+    //    T: Identifiable,
+    //{
+    //    let mut request = ListObjectsRequest::default();
+    //    let key = self.s3_key(ident, target).to_string_lossy().into_owned();
+    //    request.bucket = self.bucket.to_owned();
+    //    match self.client.list_buckets().sync() {
+    //        Ok(bucket_list) => match bucket_list.buckets {
+    //            Some(bkts) => {
+    //                for bucket in bkts {
+    //                    if bucket.name.unwrap() == request.bucket {
+    //                        match self.client.list_objects(&request).sync() {
+    //                            Ok(object_list) => match object_list.contents {
+    //                                Some(contents) => {
+    //                                    for object in contents {
+    //                                        if object.key.unwrap() == key {
+    //                                            return Ok(true);
+    //                                        }
+    //                                    }
+    //                                    return Ok(false);
+    //                                }
+    //                                None => return Ok(false),
+    //                            },
+    //                            Err(e) => {
+    //                                debug!("{:?}", e);
+    //                                return Err(Error::ObjectError(e));
+    //                            }
+    //                        }
+    //                    }
+    //                }
+    //                return Ok(false);
+    //            }
+    //            None => Ok(false),
+    //        },
+    //        Err(e) => {
+    //            debug!("{:?}", e);
+    //            Err(Error::BadBucket(e))
+    //        }
+    //    }
+    //}
+
+    // This function checks whether or not the
+    // configured bucket exists in the configured
+    // backend.
+    fn bucket_exists(&self) -> Result<bool> {
+        let artifactbucket = self.bucket.to_owned();
+        match self.client.list_buckets().sync() {
+            Ok(bucket_list) => match bucket_list.buckets {
+                Some(buckets) => {
+                    for bucket in buckets {
+                        if bucket.name.unwrap() == artifactbucket {
+                            return Ok(true);
+                        } else {
+                            return Ok(false);
```

Comment:
This logic does not seem right.. remove the else?

Reply:
You could also maybe simplify to something like:
return Ok(buckets.iter().any(|&b| b.name.unwrap() == artifactbucket)))